Add new example of ADC with RTIC using SysTick as monotonic

ceigel opened the pull request:
# Summary
[describe your changes here]

# Checklist
  - [x] `CHANGELOG.md` for the BSP or HAL updated
  - [x] All new or modified code is well documented, especially public items
  - [x] No new warnings or clippy suggestions have been introduced (see CI or check locally)

## If Adding a new Board
  - [x] Board CI added to `crates.json`
  - [x] Board is properly following "Tier 2" conventions, unless otherwise decided to be "Tier 1"

ceigel committed:
Add new example of ADC with RTIC using SysTick as monotonic

TDHolmes commented on boards/feather_m0/Cargo.toml:
```diff
@@ -28,6 +28,15 @@ optional = true
+
+[dependencies.systick-monotonic]
+version = "1.0.0"
+optional = true
```
As this is a dependency only in an example, it should go in dev-dependencies

TDHolmes replied:
Same with rtt stuff I think

ceigel replied:
@TDHolmes thank you for your suggestions. I applied them to the PR.

bradleyharden commented on boards/feather_m0/examples/adc_rtic.rs:
```diff
@@ -0,0 +1,79 @@
+    #[local]
+    struct Local {
+        adc: Adc<ADC>,
+        adc_pin: Pin<pin::PA07, Alternate<B>>,
```
It might be nice to add a pin alias to the `bsp_pins!` macro for this case.

bradleyharden commented on boards/feather_m0/examples/adc_rtic.rs:
```diff
@@ -0,0 +1,79 @@
+        let max_range = 1 << RESOLUTION_BITS;
+        let voltage = ((data as f32) * 3.3 * 2.0) / (max_range as f32);
+        rprintln!("Battery level {} - {}", voltage, data);
+        // If the LED were a local resource, the lock would not be necessary
```
What is this comment referring to? Is it a copy/paste artifact?

bradleyharden commented:
Pinging @ceigel. Did you want to pick this back up? An RTIC example for the `feather_m0` would be nice to have.

ceigel commented:
@bradleyharden sorry for the delay. I addressed the comments. I haven't tested it on hardware with the new changes. I will do that in the following days.

ceigel commented:
@bradleyharden I tested this on hardware. The PR is ready to be merged.

jbeaurivage commented:
@ceigel, Not sure if you're still interested in working on this, but I'll be happy to merge this if you want to bring the example up to speed with the current master.
